### ISSUE TITLE
x11-wm/awesome-9999: CMake Error at AutoOption.cmake:26: Value of WITH_DBUS must be one of ON/OFF/AUTO, but is yes

### ISSUE THREAD

**Issue by Manfred-Knick** [awesomeWM/awesome#2782]
Output of `awesome --version`:

# awesome --version
awesome v4.3-33-gec47abb4-dirty (Too long)
 • Compiled against Lua 5.1.5 (running with Lua 5.1)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.0

**How to reproduce the issue:**

build / compile via emerge

**Actual result:**

cmake fails ^ title

LOG:  c.f. https://bugs.gentoo.org/687376

**Comment by Elv13** [comment]
Hi,

The correct values for a cmake option are `ON` and `OFF` [1] . `AUTO` is Awesome specific.

The ebuild uses:

```makefile
src_configure() {
    # Compression of manpages is handled by portage
    local mycmakeargs=(
        -DSYSCONFDIR="${EPREFIX}"/etc
        -DCOMPRESS_MANPAGES=OFF
        -DWITH_DBUS=$(usex dbus)
        -DGENERATE_DOC=$(usex doc)
        -DAWESOME_DOC_PATH="${EPREFIX}"/usr/share/doc/${PF}
    )
    if use luajit; then
        mycmakeargs+=("-DLUA_INCLUDE_DIR=${EPREFIX}/usr/include/luajit-2.0")
        mycmakeargs+=("-DLUA_LIBRARY=${EPREFIX}/usr/$(get_libdir)/libluajit-5.1.so")
    fi
    cmake-utils_src_configure
}

```

if `$(usex dbus)` is `yes` instead of `ON`, then the error message is correct and the ebuild is wrong. The error message is from our validation code (`AutoOption.cmake`).

[1] https://cmake.org/cmake/help/v3.0/command/option.html

**Comment by psychon** [comment]
According to https://gitweb.gentoo.org/repo/gentoo.git/tree/x11-wm/awesome/awesome-9999.ebuild, Gentoo uses `usex dbus` to produce the value for `WITH_DBUS`.

According to https://devmanual.gentoo.org/ebuild-writing/eapi/index.html:

> USAGE: usex <USE flag> [true output] [false output] [true suffix] [false suffix]
> DESCRIPTION:
> If USE flag is set, echo [true output][true suffix] (defaults to "yes"),
>  otherwise echo [false output][false suffix] (defaults to "no").

So, I guess `usex dbus` has to be replaced with `usex dbus ON OFF`.

Anyway, I'd say that this is not an issue in AwesomeWM. Instead, we recently changed things to check the CMake arguments more carefully and this showed that the Gentoo ebuild has been passing in invalid values all along.

Reference to the git commit that made us more strict: c7cdb0e47910ba40793cf396dffe527878dd49c9

(But according to `cmake --help-command if`, CMake does also consider `yes` and `no` as true/false values, so I guess the ebuild did actually work fine before. It's just that the `AutoOption.cmake` that I wrote is too strict.)
